client/main.go

```diff
@@ -380,40 +387,70 @@ func main() {
 		}()
 	}
 
-	var wg sync.WaitGroup
-	wg.Add(*concurrency)
-	shutdownChannels := make([]chan struct{}, *concurrency)
+	var mainWait sync.WaitGroup
+	mainWait.Add(*connections)
 
-	for i := int(0); i < *concurrency; i++ {
+	concurrency := *connections * *streams
+	shutdownChannels := make([]chan struct{}, concurrency)
+	for c := int(0); c < concurrency; c++ {
 		shutdownChannel := make(chan struct{}, 2)
 		shutdownChannels = append(shutdownChannels, shutdownChannel)
+	}
+
+	var connOpts []grpc.DialOption
+	if *tlsTrustChainFile != "" {
+		creds, err := credentials.NewClientTLSFromFile(*tlsTrustChainFile, "")
+		if err != nil {
+			exUsage("invalid ca cert file")
```

Comment:
tioli: it might be time to widen exUsage to take an optional error. I wouldn't fix this until you need it.